src/app/shared/horizontal-carousel/horizontal-carousel.ts

```diff
@@ -0,0 +1,103 @@
+import {
+  AfterContentInit,
+  Component,
+  ContentChildren,
+  HostBinding,
+  Inject,
+  InjectionToken,
+  Input,
+  NgModule,
+  Optional,
+  QueryList,
+} from '@angular/core';
+import {CommonModule} from '@angular/common';
+import {MatIconModule} from '@angular/material/icon';
+import {MatButtonModule} from '@angular/material/button';
+import {CarouselItem} from './horizontal-carousel-directive';
+
+const CAROUSEL_WIDTH_RATIO = 0.75; // carousel width should be approximately 75% of page
```

Comment:
Hmm, this seems a _little_ specific. The carousel only works when it's a specific width? Are we able to use the native element and get the actual width instead with offsetWidth/clientWidth/innerWidth?